File: solutions/3000. Maximum Area of Longest Diagonal Rectangle/3000.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
public:
    int areaOfMaxDiagonal(const vector<vector<int>>& dimensions) {
        int maxDiagonal = 0;
        int maxArea = 0;

        for (auto& dimension : dimensions) {
            int length = dimension[0];
            int width = dimension[1];

            int diagonal = sqrt(length * length + width * width);
            int area = length * width;

            if (diagonal > maxDiagonal) {
                maxDiagonal = diagonal;
                maxArea = area;
            }
            else if (diagonal == maxDiagonal && area > maxArea) {
                maxArea = area;
            }
        }

        return maxArea;
    }
};

File: solutions/3000. Maximum Area of Longest Diagonal Rectangle/3000.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
    def areaOfMaxDiagonal(self, dimensions: List[List[int]]) -> int:
        return max((v[0] * v[0] + v[1] * v[1], v[0] * v[1]) for v in dimensions)[1]    